### PR TITLE
Read server_port from ni_grpc_device_server's stdout

### PR DESCRIPTION
**Pull request by DavidCurtiss**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

~~- [ ] I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?

Reads the ni_grpc_device_server's port from its STDOUT, instead of hard-coding it, since it now starts on a dynamic port.

### List issues fixed by this Pull Request below, if any.

[AB#2222455](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2222455)

### What testing has been done?

* Ran the NI-DCPower system tests locally.
* PR build

### CODE REVIEW

**Review by marcoskirsch — CHANGES_REQUESTED** [review]
Can you refactor this function out to a single place?

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 1]
```diff
@@ -1066,9 +1066,28 @@ def session_creation_kwargs(self):
...
+        self._proc = subprocess.Popen([str(server_exe)], stdout=subprocess.PIPE)
+
+        # Read/parse first line of output; discard the rest
+        try:
+            first_line = self._proc.stdout.readline()
+            assert first_line.startswith(b"Server listening on port "), f"Unrecognized output: {first_line}"
+            self.server_port = int(first_line.replace(b"Server listening on port ", b"").strip())
+
+            self._stdout_thread = threading.Thread(target=self._discard_output, args=(self._proc.stdout,), daemon=True)
+            self._stdout_thread.start()
```
Comment (ni-jfitzger) [inline src/nidcpower/system_tests/test_system_nidcpower.py:1081]: We're doing this just to close the process more cleanly, it looks like?

Comment (DavidCurtiss) [reply src/nidcpower/system_tests/test_system_nidcpower.py:1081]: If the process prints a bunch of stuff to STDOUT and fills up its output buffer, it'll hang on the next print if we don't empty it. It is not a problem currently, but I added it to be safe.

**Comment by DavidCurtiss** [comment]
> Can you refactor this function out to a single place?

Where to?

1. My first inclination is to generate/copy a central file, but unlike the unit tests, the system tests are run in-place in the driver-specific src dirs. Should I change that to run system tests from the generated/ folder?
2. Another option is to put it directly under src/ or in a new folder like src/shared/, and then have the test modify `sys.path` so that it can import from there.
3. A modification of option 2: have `tox` modify PYTHONPATH when calling the system tests.

**Comment by DavidCurtiss** [comment]
> Where to?

@marcoskirsch and I discussed it offline, and chose option 2. Implemented.

**Review by marcoskirsch — APPROVED** [review]

**Comment by ni-jfitzger** [comment]
travis-ci failed due to tox 4.0 releasing: https://app.travis-ci.com/github/ni/nimi-python/jobs/590575554

I'm working on a fix. #1875

**Comment by ni-jfitzger** [comment]
@DavidCurtiss you can now merge ni:master to fix the travis-ci build.

**Comment by DavidCurtiss** [comment]
> @DavidCurtiss you can now merge ni:master to fix the travis-ci build.

@ni-jfitzger Done. All checks passing now.

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
-import subprocess
+import sys
 import tempfile
-import time
+
+import grpc
+import hightime
+import pytest
+
+import nidcpower
```
Comment (ni-jfitzger) [inline src/nidcpower/system_tests/test_system_nidcpower.py:10]: This feels hacky, but I suppose it's a fair tradeoff to avoid duplicating code.

**Review by ni-jfitzger — APPROVED** [review]

**Pull request merged**